src/mscorlib/src/System/String.cs

```diff
@@ -573,10 +572,13 @@ public override bool Equals(Object obj)
         // Determines whether two strings match.
         [Pure]
         [ReliabilityContract(Consistency.WillNotCorruptState, Cer.MayFail)]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
@jamesqo We don't have the technology to broadly assess code size impact yet in CoreCLR. But you can at least look at the impact across mscorlib and the framework assemblies. Take a look at the [jitutils repo](https://github.com/dotnet/jitutils) for some tooling that @russellhadley has worked on.

The jit has some (limited) ability to determine when the call site context might lead to specializations in the callee (see for instance the logic in `fgFindJumpTargets`). It would be good to assess how that performs on the existing code. If those mechanisms are broken or limited, improving them might yield much broader benefits.

Also I have aspirations of changing the inline heuristics and I'd hate to see the framework code become dependent on what might be quirks of the current set of heuristics.